terminate called after throwing an instance of 'std::__1::bad_weak_ptr'
  what():  bad_weak_ptr
bash: line 7:  4571 Aborted                 (core dumped) ./a.out
